noxfile.py

```diff
@@ -157,6 +157,38 @@ def install_ci(session, group):
             )
 
 
+@nox.session
+def configure_coverage(session) -> None:
+    """Write a coverage config file, adding extra patterns to omit."""
+    import tomlkit
+
+    groups_str = session.posargs[0]  # first positional argument
+
+    print(groups_str)  # for debugging
+    assert isinstance(groups_str, str)  # noqa: S101 so that we don't change this away from string
+
+    if "curator" not in groups_str:
+        extra_omit_patterns = ["**/curators/*"]
+    else:
+        extra_omit_patterns = []
+
+    # Read patterns from pyproject.toml
+    base_config_path = Path("pyproject.toml")
+    with open(base_config_path) as f:
+        config = tomlkit.load(f)
+
+    # Update the omit patterns
+    base_patterns = config["tool"]["coverage"]["run"]["omit"]
+    all_patterns = base_patterns + extra_omit_patterns
+    config["tool"]["coverage"]["run"]["omit"] = all_patterns
+
+    # Write back to pyproject.toml
+    with open(base_config_path, "w") as f:
+        tomlkit.dump(config, f)
+
+    print(base_config_path.read_text())
```

Comment:
And this?

Reply:
Yeah, that's needed to check that things worked out the way they're supposed to work.

You'll not expand this step in the `test` job unless you want to debug it. That's why it's there.